### ISSUE TITLE
Focus should not stays on disabled up/down arrow key in calendar date picker control on pressing enter while selecting date

### ISSUE THREAD

**Issue by HarkeshS** [OfficeDev/office-ui-fabric-react#11746]
Focus should not stays on disabled up/down arrow key in calendar date picker control on pressing enter while selecting date

### Environment Information

OS: Win 10
Browser: IE11, Mozilla Firefox

### Describe the issue:
Focus stays on disabled up/down arrow key in calendar date picker control on pressing enter while selecting date. Focus should not be there in case of up/down keys are disabled. See below snaps

![DatePicker_IE11_Issue](https://user-images.githubusercontent.com/53938674/72790028-cd75e400-3c5a-11ea-91c3-6cc31c6cb0db.png)

![DatePicker_Firefox_Issue](https://user-images.githubusercontent.com/53938674/72790117-f5654780-3c5a-11ea-8b9c-d8a025814e8d.png)

### Please provide a reproduction of the issue in a codepen:

https://developer.microsoft.com/en-us/fabric#/controls/web/datepicker
Repro steps on above link with (DatePicker with date boundary (minDate, maxDate))

#### Actual behavior:

Focus stays on disabled up/down arrow key in calendar date picker control on pressing enter while selecting date

#### Expected behavior:

Focus should not stays on disabled up/down arrow key in calendar date picker control on pressing enter while selecting date
Working fine in Edge and Chrome browser, issue only in Firefox and IE11

**Comment by khmakoto** [comment]
Hi @HarkeshS, I can confirm that I can reproduce this behavior. However, it is weird that it is only reproducing on IE and Firefox. We are looking onto dropping IE support now that the Chromium based Edge is available in older versions of Windows. Furthermore, upon some inspection it seems to me that we're doing the correct thing in our control, so maybe this is an issue with Firefox itself.

@lorejoh12 what is your take on this? I suspect it is an issue with Firefox that should be filed to them instead of an issue with our control.

**Comment by lorejoh12** [comment]
This looks like an issue in Firefox. In Edge and Chrome, when an element has focus on it and gets disabled, the focus moves to the previously focusable element (as if you had hit shift tab). In Firefox, for whatever reason, the focus stays on the now-invalid button until something moves it. Once you tab forwards and backwards, focus skips over the control as expected, so it's only the initial focus state that's broken.

It would be nice if Firefox changed this behavior to match other browsers, but from my searching I wasn't able to find any bugs logged or documentation of this behavior. It's possible for us to handle this ourselves in Fabric by manually changing the focus as well once we detect that the button is going to be disabled.

**Comment by khmakoto** [comment]
@lorejoh12 thanks for your input on this issue!

@HarkeshS, would you be willing to open give feedback to Firefox explaining this disparity? While we could manually do it like John says, I would be against doing non-standard fixes for things the browser should do itself.
